### PR TITLE
Feature/add scan limit

### PR DESCRIPTION
**Pull request by mitra42**
Merges guts of /scan and /sync

### CODE REVIEW

File: contacts.py [hunk 1]
```diff
@@ -335,18 +335,25 @@ def _map_over_matching_contacts(self, prefix, ids, since, now, start_pos=0):
...
+    def map_over_prefixes(self, prefixes, since, now):
         """
         Return relative file paths that match the prefix and are between the times
+        If no prefix is provided then can shortcut by using the sorted_list_by_time_and_serial_number BUT returns (floating_point_seconds, serial)
         """
-        yield from self._map_over_matching_contacts(key, self.items, since, now)
-        return
+        if prefixes is None:
+            l = self.sorted_list_by_time_and_serial_number
+            # TODO-DAN TODO-120 this feels wrong but cant find alternative that works
```
Comment (mitra42) [inline contacts.py:350]: Lets get this right before merging !

File: contacts.py [hunk 2]
```diff
@@ -411,14 +418,20 @@ def bounds(self):
...
-            key = obj.object
-            for (floating_time, serial_number) in self.get_bottom_level_from_key(key)[key]:
-                if _good_date(floating_time, since, now):
-                    file_name = '%s:%f:%d.data' % (key, floating_time, serial_number)
-                    yield FSBackedThreeLevelDict.get_file_path_from_file_name(file_name)
+    # return floating_time_and_serial
+    def map_over_bounding_boxes(self, bounding_boxes, since, now):
+        if bounding_boxes is None:
+            l = self.sorted_list_by_time_and_serial_number
+            # TODO-DAN TODO-120 this feels wrong but cant find alternative that works
```
Comment (mitra42) [inline contacts.py:425]: See above same isue

**Comment by mitra42** [comment]
Closes #120

File: contacts.py [hunk 3]
```diff
@@ -663,12 +657,19 @@ def _status_or_sync(self, prefixes, bounding_boxes, since, now, number_to_return
...
+                            self.contact_dict.map_over_prefixes(prefixes, since, now)))
+        else:
+            l = self.contact_dict.sorted_list_by_time_and_serial_number
+            data.update(map(lambda item: (item[0], item[1], self.contact_dict),
+                            l[l.bisect_left((since, 0)):l.bisect_left((now, 0))]))
+        if bounding_boxes is None:
+            data.update(map(lambda item: (item[0], item[1], self.spatial_dict),
+                            self.spatial_dict.map_over_bounding_boxes(bounding_boxes, since, now)))
         else:
             l = self.contact_dict.sorted_list_by_time_and_serial_number
```
Comment (danaronson) [inline contacts.py:671]: uh, should this be spatial_dict?

Comment (mitra42) [reply contacts.py]: Yep - fixed and pushed

**Comment by mitra42** [comment]
Closes #120

**Pull request merged**